sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -179,7 +183,12 @@ sycl::marray<bfloat16, N> fmin(sycl::marray<bfloat16, N> x,
     std::memcpy(&res[i * 2], &partial_res, sizeof(uint32_t));
   }
 
-  if constexpr (N % 2) {
+#if __cplusplus >= 201703L
+  if constexpr (N % 2)
+#else
+  if (N % 2)
+#endif // __cplusplus >= 201703L
+  {
```

Comment:
why do we need to be this explicit that this is a constexpr? Is this a performance critical place? Why not just remove constexpr?

Reply:
I think that all math functions count as performance critical places since they will often be called many many times when used. I think that we could remove constexpr here since at least at O3 there is no difference in the final asm code whether or not constexpr is explicitly used. However I was requested in the review to use c++17 where possible.

Do you want me to remove constexpr usage here?

Reply:
I'd remove it, but I'll let you to make final call.

Reply:
OK I've removed it. Thanks